### PR TITLE
Add azure_network_interface(s) resources

### PR DESCRIPTION
**Pull request by omerdemirok**
Signed-off-by: Omer Demirok <odemirok@chef.io>

### Description

Add fully backward compatible `azure_network_interface(s)` resources.  

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_network_interface.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+| ip_configurations       | A list of [IPConfigurations](https://docs.microsoft.com/en-us/rest/api/virtualnetwork/networkinterfaceipconfigurations/get#networkinterfaceipconfiguration) of the network interface. |
+| private_ip              | The private IP address of the interrogated network interface's first IP configuration. |
+| has_private_address_ip? | Indicates whether the interrogated network interface's first IP configuration has a private IP address. |
+| public_ip               | The public IP address of the interrogated network interface's first IP configuration. |
+| has_public_address_ip?  | Indicates whether the interrogated network interface's first IP configuration has a public IP address. |
+For properties applicable to all resources, such as `type`, `name`, `id`, `properties`, refer to [`azure_generic_resource`](azure_generic_resource.md#properties).
+
+Also, refer to [Azure documentation](https://docs.microsoft.com/en-us/rest/api/virtualnetwork/networkinterfaces/get#networkinterface) for other properties available. 
+Any attribute in the response may be accessed with the key names separated by dots (`.`).
```
Comment (rmoles) [inline docs/resources/azure_network_interface.md:67]: I wonder if we should add an example to the end of this statement
```
Any attribute in the response may be accessed with the key names separated by dots (`.`). eg. `properties.<attribute>`
```

**Review by rmoles — APPROVED** [review]
One minor comment. Happy to merge either way though. Thanks, @omerdemirok.

File: docs/resources/azure_network_interface.md (hunk 1 above)
Comment (omerdemirok) [reply docs/resources/azure_network_interface.md:67]: Good catch, I'll update it.
I try to give such an example in the examples section usually.

Comment (omerdemirok) [reply docs/resources/azure_network_interface.md]: Updated.

**Pull request merged**